pandas/core/frame.py

```diff
@@ -1925,7 +1925,7 @@ def _create_data_for_split_and_tight_to_dict(
     def to_dict(
         self,
         orient: Literal["dict", "list", "series", "split", "tight", "index"] = ...,
-        into: type[dict] = ...,
+        into: type[dict] | dict = ...,
```

Comment:
Is it possible to use `type[MutableMapping] | MutableMapping`? The documentation says any subclass of `abc.Mapping` (but it needs to be mutable).

Reply:
Yes, sure. How about (would _typing.py be the best place?)

```
MutableMappingT = TypeVar('MutableMappingT', bound=MutableMapping)
```

and then use `... into: type[MutableMappingT] | MutableMappingT) -> MutableMappingT:`

This should then specify the return type matches the type given for `into` (rather than any `MutableMapping` subclass)

Reply:
Yes, that sounds good!

Reply:
Done, though a bit more involved than I planned -- a generic type with a default triggered https://github.com/python/mypy/issues/3737. The workaround is to use overloads and not type the implementation, but manual tests (with pytype) show that the return type is still correctly inferred:

```
reveal_type(df.to_dict(into={})) # gives dict
reveal_type(df.to_dict(into=collections.defaultdict(list)))  # gives collections.defaultdict[nothing, list]
```